### PR TITLE
Reduce raw types usage

### PR DESCRIPTION
**Pull request by rsynek**
Also:
- renames `resolvedActiveThreadCount` to `resolveActiveThreadCount`
- fixes error message by passing correct instance to `ConfigUtils.newInstance`.

### CODE REVIEW

**Comment by triceo** [comment]
Although it may be worth getting rid of some of the Sonar warnings.

**Comment by rsynek** [comment]
> Although it may be worth getting rid of some of the Sonar warnings.

Right, these were false positives:
- a licence header detected as a code duplication
- fragments of error messages, again, marked as code duplication

after removing them, the maintainability is back to A (although the check would need to rerun). Merging.

File: optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhaseFactory.java [hunk 1]
```diff
@@ -76,13 +77,13 @@ public DefaultPartitionedSearchPhaseFactory(
         return phase;
     }
 
-    private SolutionPartitioner buildSolutionPartitioner() {
+    private SolutionPartitioner<Solution_> buildSolutionPartitioner() {
         if (phaseConfig.getSolutionPartitionerClass() != null) {
-            SolutionPartitioner<?> solutionPartitioner = ConfigUtils.newInstance(this,
-                    "solutionPartitionerClass", phaseConfig.getSolutionPartitionerClass());
+            SolutionPartitioner<?> solutionPartitioner =
+                    ConfigUtils.newInstance(phaseConfig, "solutionPartitionerClass", phaseConfig.getSolutionPartitionerClass());
```
Comment (yurloc) [inline optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhaseFactory.java:83]: Why have you replaced `this` with `phaseConfig`?

Comment (rsynek) [reply optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhaseFactory.java:83]: If you take a look at the `ConfigUtils.newInstance()`, the reference is used to properly communicate an error during instantiation of the `SolutionPartitionerClass`, which is a property of the `phaseConfig` object.

**Pull request merged**